src/ResourceManagement/AppService/FunctionAppImpl.cs

```diff
@@ -154,17 +160,31 @@ internal override async Task<Models.SiteInner> SubmitAppSettingsAsync(SiteInner
             }
         }
 
+        private bool IsConsumptionAppServicePlan(PricingTier pricingTier)
+        {
+            SkuDescription description = pricingTier.SkuDescription;
+            return !(description.Tier.Equals("Basic", StringComparison.OrdinalIgnoreCase)
+                || description.Tier.Equals("Standard", StringComparison.OrdinalIgnoreCase)
+                || description.Tier.Equals("Premium", StringComparison.OrdinalIgnoreCase));
+        }
+
         public override async Task<IFunctionApp> CreateAsync(CancellationToken cancellationToken = default(CancellationToken), bool multiThreaded = true)
         {
-            if (Inner.ServerFarmId == null)
+            if (IsInCreateMode)
             {
-                WithNewConsumptionPlan();
-            }
-            if (currentStorageAccount == null && storageAccountToSet == null && storageAccountCreatable == null)
+                if (Inner.ServerFarmId == null)
+                {
+                    WithNewConsumptionPlan();
+                }
+                if (currentStorageAccount == null && storageAccountToSet == null && storageAccountCreatable == null)
+                {
+                    WithNewStorageAccount(SdkContext.RandomResourceName(Name, 20).Replace("-", String.Empty), Storage.Fluent.Models.SkuName.StandardGRS);
+                }
+            } else
             {
-                WithNewStorageAccount(SdkContext.RandomResourceName(Name, 20).Replace("-", String.Empty), Storage.Fluent.Models.SkuName.StandardGRS);
+                // for Update+Apply, skip check null and create new app service plan and storage account, use existing ones if caller does not specify otherwise
             }
-            return await base.CreateAsync(cancellationToken);
+            return await base.CreateAsync(cancellationToken, multiThreaded);
```

Comment:
What's the reason to add the multiThreaded here?

Reply:
method signature is
`public override async Task<IFunctionApp> CreateAsync(CancellationToken cancellationToken = default(CancellationToken), bool multiThreaded = true)`

so I guess previous base call miss that multiThreaded param.